using_restricted_array.rb

```diff
@@ -18,36 +19,88 @@ def length(array)
 
 # Prints each integer values in the array
 def print_array(array)
-  puts "NOT IMPLEMENTED"
+  i = 0
+  until array[i] == nil
+    print "#{array[i]}"
+    if array[i+1] == nil
+      print "\n"
+    else
+      print ', '
+    end
+    i += 1
+  end
 end
 
 # For an unsorted array, searches for 'value_to_find'.
 # Returns true if found, false otherwise.
 def search(array, length, value_to_find)
-  puts "NOT IMPLEMENTED"
+  i = 0
+  until i == length
+    if array[i] == value_to_find
+      return true
+    end
+    i += 1
+  end
+  return false
 end
 
 # Finds and returns the largest integer value the array
 # Assumes that the array is not sorted.
 def find_largest(array, length)
-  puts "NOT IMPLEMENTED"
+  i = 0
+  max = array[0]
+  until i == length
+    if array[i] > max
+      max = array[i]
+    end
+    i += 1
+  end
+  max
 end
 
 # Finds and returns the smallest integer value in the array
 # Assumes that the array is not sorted.
 def find_smallest(array, length)
-  puts "NOT IMPLEMENTED"
+  i = 0
+  min = array[0]
+  until i == length
+    if array[i] < min
+      min = array[i]
+    end
+    i += 1
+  end
+  min
 end
 
 # Reverses the values in the integer array in place
 def reverse(array, length)
-  puts "NOT IMPLEMENTED"
+  a = 0
+  b = length - 1
+  until a >= b
+    array[a] = array[a] + array[b]
```

Comment:
Please note that this will only work for small values of integers. It's usually safer to use the swap mechanic using a temp variable.